core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/common/AbstractUserStoreManager.java

```diff
@@ -12721,6 +12726,13 @@ public String getUserIDFromUserName(String userName) throws UserStoreException {
         if (StringUtils.isEmpty(userID)) {
             if (isUniqueUserIdEnabledInUserStore(userStore)) {
                 userID = doGetUserIDFromUserNameWithID(userName);
+                if (StringUtils.isEmpty(userID)) {
+                    if (log.isDebugEnabled()) {
+                        log.debug(String.format("User with userName %s is not available in cache or database.",
+                                userName));
```

Comment:
No need to using String.format since we are passing only one param here.

Reply:
Here apache commons is used as the logging method and it does not support parameterized logging. Check the SLF4J based logging methods which allows parameterized logging.

Reply:
Fixed by - https://github.com/wso2/carbon-kernel/pull/3575/commits/1802bad2420debc4069d2b86d1d632b09c155d5c